interfaces/Portalicious/src/app/pages/project-registration-personal-information/project-registration-personal-information.page.ts

```diff
@@ -55,59 +52,80 @@ export class ProjectRegistrationPersonalInformationPageComponent {
   readonly registrationApiService = inject(RegistrationApiService);
   readonly projectApiService = inject(ProjectApiService);
 
+  project = injectQuery(this.projectApiService.getProject(this.projectId));
   projectAttributes = injectQuery(
     this.projectApiService.getProjectAttributes({
       projectId: this.projectId,
       includeProgramRegistrationAttributes: true,
       includeTemplateDefaultAttributes: false,
     }),
   );
-
   registration = injectQuery(
     this.registrationApiService.getRegistrationById(
       this.projectId,
       this.registrationId,
     ),
   );
 
-  project = injectQuery(this.projectApiService.getProject(this.projectId));
+  attributeList = computed<PersonalInformationAttribute[]>(() => {
+    const genericAttributes =
+      this.genericAttributeNamesForPersonalInformation();
 
-  attributeList = computed<DataListItem[]>(() => {
-    const list: DataListItem[] = [];
-    if (!this.projectAttributes.isSuccess() || !this.registration.isSuccess()) {
-      return list;
-    }
+    const projectSpecificAttributes = (
+      this.projectAttributes.data() ?? []
+    ).filter(
+      (attribute) =>
+        // both of these are handled elsewhere we don't want to duplicate them here
+        attribute.name !== 'fullName' && attribute.name !== 'phoneNumber',
+    );
 
-    const genericAttributeNames =
-      this.genericAttributeNamesForPersonalInformation();
-    for (const attributeName of genericAttributeNames) {
-      const value = getValueForGenericAttribute(
-        this.registration.data()[attributeName],
-        attributeName,
-      );
-      const genericAttributeListItem = getGenericAttributeDataListItem(
-        attributeName,
-        value,
-      );
-      list.push(genericAttributeListItem);
-    }
-    for (const attribute of this.projectAttributes.data()) {
-      const value = this.registration.data()[attribute.name] as unknown;
-      const dataListItem = projectAttributeToDataListItem(attribute, value);
-      if (dataListItem) {
-        list.push(dataListItem);
-      }
-    }
-    return list;
+    return [
+      {
+        name: 'name',
+        label: $localize`:@@registration-full-name:Name`,
+        value: this.registration.data()?.name,
+        type: RegistrationAttributeTypes.text,
+      },
+      ...genericAttributes.map((attributeName) => {
+        const value = getValueForGenericAttribute(
+          this.registration.data()?.[attributeName],
+          attributeName,
+        );
+        const type = getGenericAttributeType(attributeName);
+
+        return {
+          name: attributeName,
+          label: ATTRIBUTE_LABELS[attributeName],
+          value,
+          type,
+        };
+      }),
+      ...projectSpecificAttributes.map(({ name, label, type }) => {
+        return {
+          name,
+          label,
+          value: this.registration.data()?.[name],
+          type,
+        };
+      }),
+    ];
   });
 
+  dataList = computed<DataListItem[]>(() =>
```

Comment:
Separating `attributeList` from `dataList` is laying the groundwork for editing personal information (in which we will need `attributeList` but not `dataList`).